File: src/main/java/com/mini/credit/dto/membre/MembreCreateRequest.java
package com.mini.credit.dto.membre;

import com.mini.credit.enums.Sexe;
import com.mini.credit.enums.StatutMembre;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;

@Data
public class MembreCreateRequest {

    @NotBlank
    private String nom;

    private String postnom;
    private String prenom;

    @NotBlank
    private String nomComplet;

    private Sexe sexe;
    private LocalDate dateNaissance;

    private String telephonePrincipal;
    private String telephoneSecondaire;

    private String adresse;
    private String ville;
    private String commune;
    private String quartier;

    private String professionActivite;
    private String lieuActivite;

    @NotBlank
    private String codeMembre;

    @NotNull
    private LocalDate dateAdhesion;

    @NotNull
    private StatutMembre statut;

    private Long agentId;
    private Long siteId;
}

File: frontend/src/app/features/membres/models/membre-create-request.ts
import { Sexe } from "../../../shared/enums/sexe.enum";


export interface MembreCreateRequest {
  nom: string;

  postnom?: string;
  prenom?: string;

  sexe?: Sexe;
  dateNaissance?: string;

  telephonePrincipal?: string;
  telephoneSecondaire?: string;

  adresse?: string;
  ville?: string;
  commune?: string;
  quartier?: string;

  professionActivite?: string;
  lieuActivite?: string;
  sourceInscription?: string;

  email: string;

  siteId: number;
  agentId?: number;

  dateAdhesion: string;

  observation?: string;
}